modules/HapticGlove_module/src/HapticGloveModule.cpp

```diff
@@ -0,0 +1,200 @@
+        yError() << m_logPrefix << "unable to get a string from a searchable";
+        return false;
+    }
+    setName(name.c_str());
+
+    yarp::os::Bottle& generalOptions = rf.findGroup("GENERAL");
+    // get the period
+    m_dT = generalOptions.check("samplingTime", yarp::os::Value(0.1)).asDouble();
+
+    // robot name:
+    m_robot = generalOptions.check("robot", yarp::os::Value("icubSim")).asString();
+
+    // check which hand to use
+    m_useLeftHand = generalOptions.check("useLeftHand", yarp::os::Value(1)).asBool();
+    m_useRightHand = generalOptions.check("useRightHand", yarp::os::Value(1)).asBool();
+    yInfo() << m_logPrefix << "use the left hand: " << m_useLeftHand;
+    yInfo() << m_logPrefix << "use the right hand: " << m_useRightHand;
+
+    // initialize the left hand teleoperation
+    if (m_useLeftHand)
+    {
+        yarp::os::Bottle& leftFingersOptions = rf.findGroup("LEFT_FINGERS_RETARGETING");
+        leftFingersOptions.append(generalOptions);
+
+        m_leftHand = std::make_unique<HapticGlove::Teleoperation>();
+        if (!m_leftHand->configure(leftFingersOptions, getName(), false))
+        {
+            yError() << m_logPrefix
+                     << "unable to initialize the left hand bilateral teleoperation.";
+            return false;
+        }
+    }
+
+    // initialize the right hand teleoperation
+    if (m_useRightHand)
+    {
+        yarp::os::Bottle& rightFingersOptions = rf.findGroup("RIGHT_FINGERS_RETARGETING");
+        rightFingersOptions.append(generalOptions);
+
+        m_rightHand = std::make_unique<HapticGlove::Teleoperation>();
+        if (!m_rightHand->configure(rightFingersOptions, getName(), true))
+        {
+            yError() << m_logPrefix
+                     << "unable to initialize the right hand bilateral teleoperation.";
+            return false;
+        }
+    }
+
+    // update the end of the configuration time step
+    double timeConfigurationEnd = yarp::os::Time::now();
+
+    if (m_useLeftHand)
+    {
+        m_leftHand->setEndOfConfigurationTime(timeConfigurationEnd);
+    }
+
+    if (m_useRightHand)
+    {
+        m_rightHand->setEndOfConfigurationTime(timeConfigurationEnd);
+    }
+
+    yInfo() << m_logPrefix << "configuration is done. ";
+    m_state = HapticGloveFSM::Preparing;
+
+    return true;
+}
+
+double HapticGloveModule::getPeriod()
+{
+    return m_dT;
+}
+
+bool HapticGloveModule::close()
+{
+    yInfo() << m_logPrefix << "trying to close.";
+    if (m_useLeftHand)
+    {
+        if (!m_leftHand->close())
+        {
+            yError() << m_logPrefix << "unable to close left hand teleoperation.";
+            return false;
+        }
+    }
+
+    if (m_useRightHand)
+    {
+        if (!m_rightHand->close())
+        {
+            yError() << m_logPrefix << "unable to close right hand teleoperation.";
+            return false;
+        }
+    }
+
+    std::this_thread::sleep_for(std::chrono::milliseconds(100)); // wait for 100ms.
```

Comment:
Why is this needed?

Reply:
Removed from here. In any case, the there is sleep in the close method of teleoperation class right after sending `stopHapticFeedback()` in order to be sure the stop haptic command is sent before closing the module, otherwise the glove may continue to provide the haptic feedback according to the last sent command.

Reply:
I see. The important thing is to write why there is that delay so that it is understandable in the future.